notation.go

```diff
@@ -177,10 +177,19 @@ func Verify(ctx context.Context, verifier Verifier, repo registry.Repository, op
 			}
 			numOfSignatureProcessed++
 			// get signature envelope
-			sigBlob, _, err := repo.FetchSignatureBlob(ctx, sigManifestDesc)
+			sigBlob, sigDesc, err := repo.FetchSignatureBlob(ctx, sigManifestDesc)
 			if err != nil {
 				return ErrorSignatureRetrievalFailed{Msg: fmt.Sprintf("unable to retrieve digital signature with digest %q associated with %q from the registry, error : %v", sigManifestDesc.Digest, artifactRef, err.Error())}
 			}
+			// opts.SignatureMediaType is non-emtpy but does not match the
+			// sigDesc.MediaType, skip the current signature.
+			if opts.SignatureMediaType != "" && opts.SignatureMediaType != sigDesc.MediaType {
+				continue
+			}
+			// Otherwise, use the MediaType fetched from repo
+			opts.SignatureMediaType = sigDesc.MediaType
```

Comment:
A little confusing. Correct me if I'm wrong.
Suppose we have 2 sigantures and opts.SignatureMediaType is empty:
1. when processing the first one `sigCose1`:  opts.SignatureMediaType = "application/cose"
2. when processing the second one `sigJWS2`: this signature will be skipped.
Because opts.SignatureMediaType is `empty`, user don't want to skip any signature type.

Reply:
Code updated.

Reply:
Is it possible that the user reuses the `VerifyOptions` but the user doesn't know we changed the `opts.SignatureMediaType` which causing a issue?